src/main/java/com/cst438/controller/EnrollmentController.java

```diff
@@ -1,42 +1,95 @@
-        //  remove the following line when done
-        return null;
+        List<Enrollment> enrollments = enrollmentRepository
+                .findEnrollmentsBySectionNoOrderByStudentName(sectionNo);
+        List<EnrollmentDTO> dlist = new ArrayList<>();
+        for (Enrollment e : enrollments) {
+            dlist.add(new EnrollmentDTO(
+                    e.getEnrollmentId(),
+                    e.getGrade(),
+                    e.getStudent().getId(),
+                    e.getStudent().getName(),
+                    e.getStudent().getEmail(),
+                    e.getSection().getCourse().getCourseId(),
+                    e.getSection().getCourse().getTitle(),
+                    e.getSection().getSecId(),
+                    e.getSection().getSectionNo(),
+                    e.getSection().getBuilding(),
+                    e.getSection().getRoom(),
+                    e.getSection().getTimes(),
+                    e.getSection().getCourse().getCredits(),
+                    e.getSection().getTerm().getYear(),
+                    e.getSection().getTerm().getSemester()));
+        }
+        return dlist;
     }
 
-    // instructor uploads enrollments with the final grades for the section
+    // instructor uploads  final grades for the section
     // user must be instructor for the section
     @PutMapping("/enrollments")
     public void updateEnrollmentGrade(@RequestBody List<EnrollmentDTO> dlist) {
+        for (EnrollmentDTO d : dlist) {
+            Enrollment e = enrollmentRepository.findById(d.enrollmentId()).orElse(null);
+            if (e==null) {
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "enrollment not found "+d.enrollmentId());
+            } else {
+                e.setGrade(d.grade());
+            }
 
-        // TODO
-
-        // For each EnrollmentDTO in the list
-        //  find the Enrollment entity using enrollmentId
-        //  update the grade and save back to database
+            EnrollmentDTO enrollmentDTO = new EnrollmentDTO(
+                    e.getEnrollmentId(),
+                    e.getGrade(),
+                    e.getStudent().getId(),
+                    e.getStudent().getName(),
+                    e.getStudent().getEmail(),
+                    e.getSection().getCourse().getCourseId(),
+                    e.getSection().getCourse().getTitle(),
+                    e.getSection().getSecId(),
+                    e.getSection().getSectionNo(),
+                    e.getSection().getBuilding(),
+                    e.getSection().getRoom(),
+                    e.getSection().getTimes(),
+                    e.getSection().getCourse().getCredits(),
+                    e.getSection().getTerm().getYear(),
+                    e.getSection().getTerm().getSemester()
+            );
 
+            registrarService.updateEnrollmentGrade(enrollmentDTO);
+        }
```

Comment:
Correctly packages the enrollment information into the enrollmentDTO to send in a message to the registrar to update the final grade of the student.